### ISSUE TITLE
Completion on confirmation

### ISSUE THREAD

**Issue by yueyingjuesha** [hrsh7th/cmp-cmdline#42]
I am wondering if there is a configuration that can auto select the first entry whenever I start typing characters
For example I currently use 
```code
  completion = { completeopt = 'menu,menuone,noinsert' }, -- pre select the first one
```
In the feature to do the trick, but it seems like when I hit <CR> on the pre select item, it doesn't auto expand to the full entry, just the character I entered, which is different from what I see in normal editor area

**Comment by appelgriebsch** [comment]
Same here. Have to tab twice to get the single item selected from the list, which is a bit annoying.

**Comment by lkschu** [comment]
Same problem.

**Comment by junn661** [comment]
You must set

**Comment by yueyingjuesha** [comment]
> You must set
> 
> ```
> mapping = {                    
>     ['<CR>'] = cmp.mapping.confirm({ select = true })
>   }
> ```

No, setting this configuration only works on editor area, it doesn't work on the cmd line status line

**Comment by jnguk** [comment]
How about this?
```
    mapping = {                
        ['<CR>'] = cmp.mapping(cmp.mapping.confirm({ select = true }), { 'i', 'c' })      
    },    
```

**Comment by thibthib18** [comment]
The above works fine. 
However, for very short (and frequent) commands such as `:q`, `:w`, `:wq` or `:qa`, the completion is a bit annoying.
Is there a way to disable completion when the input starts with `w` or `q`? Or maybe set a min character limit?

**Comment by SirWrexes** [comment]
The best here would be to have completion be aware of shorthands available for commands. It should show `w` just like it sows `write`.
Also I found out that you can `<C-CR>` to confirm whatever is in your command line without triggering completion.
